pypeit/core/flexure.py

```diff
@@ -55,68 +59,223 @@ def spat_flexure_shift(sciimg, slits, debug=False, maxlag = 20):
 
     Args:
         sciimg (`numpy.ndarray`_):
+            Science image
         slits (:class:`pypeit.slittrace.SlitTraceSet`):
+            Slits object
+        bpm (`numpy.ndarray`_, optional):
+            Bad pixel mask (True = Bad)
         maxlag (:obj:`int`, optional):
             Maximum flexure searched for
+        sigdetect (:obj:`float`, optional):
+            Sigma threshold above fluctuations for the slit detection
+            in the collapsed sobel image
+        debug (:obj:`bool`, optional):
+            Run in debug mode
+        qa_outfile (:obj:`str`, optional):
+            Path to the output file where the QA is saved.  If None, the QA is not generated.
+        qa_vrange (:obj:`tuple`, optional):
+            Tuple with the vmin and vmax values for the imshow plot in the QA. If None, the
+            vmin and vmax values are calculated from the data.
 
     Returns:
         float:  The spatial flexure shift relative to the initial slits
 
     """
+    msgs.info("Measuring spatial flexure")
+    # TODO: should we use initial or tweaked slits?
     # Mask -- Includes short slits and those excluded by the user (e.g. ['rdx']['slitspatnum'])
     slitmask = slits.slit_img(initial=True, exclude_flag=slits.bitmask.exclude_for_flexure)
 
     _sciimg = sciimg if slitmask.shape == sciimg.shape \
-                else arc.resize_mask2arc(slitmask.shape, sciimg) 
-    onslits = slitmask > -1
-    corr_slits = onslits.astype(float).flatten()
-
-    # Compute
-    mean_sci, med_sci, stddev_sci = stats.sigma_clipped_stats(_sciimg[onslits])
-    thresh =  med_sci + 5.0*stddev_sci
-    corr_sci = np.fmin(_sciimg.flatten(), thresh)
+                else arc.resize_mask2arc(slitmask.shape, sciimg)
+
+    # create sobel images of both slitmask and the science image
+    sci_sobel, _ = trace.detect_slit_edges(_sciimg, bpm=bpm)
+    slits_sobel, _ = trace.detect_slit_edges(slitmask, bpm=bpm)
+    # collapse both sobel images along the spectral direction
+    sci_smash, _, _ = sigma_clipped_stats(sci_sobel, axis=0, mask=bpm)
```

Comment:
Does this work well in cases where the slit edges are significantly tilted wrt detector columns?

Reply:
Yes, it does seems to work well. I was worried too at first, but if the slits edges are tilted for the trace image they will be tilted also for the science image.